geopandas/geodataframe.py

```diff
@@ -586,8 +589,27 @@ def from_features(cls, features, crs=None, columns=None):
             }
             # load properties
             row.update(feature["properties"])
+            if "id" in feature:
+                row.update({"id": feature["id"]})
```

Comment:
To avoid the issue I just described, we could maybe keep track of the "id" feature in a separate list (instead of updating `row`), and then afterwards use that list to create an Index and set that.